Actions/AL-Go-Helper.ps1

```diff
@@ -469,13 +482,18 @@ function ReadSettings {
                 }
             }
             catch {
-                throw "Settings file $settingsFile, is wrongly formatted. Error is $($_.Exception.Message)."
+                throw "Settings file $settingsFile, is wrongly formatted. Error is $($_.Exception.Message).`n$($_.ScriptStackTrace)"
             }
         }
     }
 
     if ($settings.githubRunner -eq "") {
-        $settings.githubRunner = $settings."runs-on"
+        if ($settings."runs-on" -like "ubuntu-*") {
+            $settings.githubRunner = "windows-latest"
```

Comment:
The other way around?

Reply:
No,

"runs-on" is the runner used for all jobs (except for build jobs)
GitHubRunner is the runner used for build jobs (default is to use runs-on)
Build jobs doesn't support Ubuntu.

Reply:
AL-Go doesn't support macos.
Partners will set runs-on to self-hosted in order to use the self-hosted for all jobs.
It is only if they set ubuntu that we won't use ubuntu for GitHubRunner (which is the build job runner)

Reply:
If _runs-on_ is _self-hosted_, then _self-hosted_ is used for all jobs.
If _runs-on_ is _ubuntu-*_ then, the build jobs will be on _windows-latest_ and the non-build jobs: on _ubuntu-*_.

Aren't these fallback settings hard to follow and maintain?

Hear me out: _githubrunner_ defaults to _runs-on_, _runs_on_ defaults to _windows-latest_. Period.
Worst case, partners have to set up one more setting.

Imagine in X time from now, if we can actually build BC on Linux, how this fallback won't make sense.

Reply:
If runs-on is self-hosted, then self-hosted is used for all jobs.
If runs-on is ubuntu-* then we will use ubuntu for all the jobs possible (and windows if the job cannot run Linux)

Re: 
_Imagine in X time from now, if we can actually build BC on Linux, how this fallback won't make sense._

Which is exactly the reason why they should NOT set gitHubRunner to Windows-latest.

If they set runs-on to Ubuntu - we will run everything (possible) on Ubuntu.
and in x months from now where we can run builds on Linux - we will do that.
If everybody needs to add a setting to set GitHubRunner Windows-Latest, then they will have to change settings again when we support Linux for builds as well.